### DISCUSSION TITLE
If pion webrtc can implement some meida Qos policy?

### DISCUSSION THREAD

**Discussion by xianchuhan** [pion/webrtc#1795] [Ideas]
Including ARQ,fec,nack, such as adaptive streaming,and so on

**Comment by Sean-Der** [comment]
Thanks for checking out Pion! This is an ever evolving topic, but this is what we have today. If anyone is interested in updates on this topic in the future feel free to mention me and will update it :) 

---

The WebRTC protocol doesn't have any ARQ type messages.

We are working on FlexFEC and RED. I don't have any timelines for when it will be done.

Pion today provides NACKs. You can also use PLIs and FIRs.

Pion also provides Receiver Reports. Anyone who is sending media to a Pion instance can implement adaptive streaming.

We are working on adding a congestion controller so that Pion can provide estimates. No estimates either. https://github.com/pion/interceptor/issues/25
